packages/babel-types/package.json

```diff
@@ -15,6 +15,11 @@
   },
   "main": "lib/index.js",
   "types": "lib/index.d.ts",
+  "typesVersions": {
+    ">=3.7": {
+      "lib/*.d.ts": ["lib/*.ts37.d.ts"]
```

Comment:
nit: I think we can swap the name of `ts37.d.ts` with `index.d.ts`. So `index.d.ts` always align to latest TS, and we can have `lib/legacy.d.ts` as fallback.
```json
{
  "types": "lib/legacy.d.ts",
  "typesVersions": {
    ">=3.7": {
      "*": "lib/index.d.ts"
    }
}
```

Reply:
The thing is, importing `lib/index.js` directly would cause old **TypeScript** versions to try to load `lib/index.d.ts`, but that would then have the **TypeScript 3.7** syntax.

Using `index.ts3.7.d.ts` also ensures that we can easily add `index.tsX.Y.d.ts` (e.g.: `index.ts4.0.d.ts`) in the future.

Reply:
We will disable submodule imports in Babel 8. And we could drop ts <3.0 support, too. So `tsc` will always respect `typeVersions` and `types` and we can enforce that `lib/index.d.ts` always aligns to latest TS while `typeVersions` serves a fallback for older `ts>=3` versions.

Reply:
`typesVersions` is only supported since **TypeScript 3.1**.